apps/snapodds/src/services/auth/auth.service.ts

```diff
@@ -23,8 +22,8 @@ export class AuthService {
     this.accessTokenUpdated.next(this.accessToken);
   }
 
-  refreshAccessToken(): Observable<string | undefined> {
-    if (this.hasValidAccessToken()) {
+  requireAccessToken(): Observable<string> {
+    if (this.accessToken != undefined && !this.isTokenExpired()) {
```

Comment:
No, it can not (at least I don't know how). I have changed the generic type of return value from Observable<string | undefined> to Observable<string> (because it should never happen that we will have an undefined to be pushed into this Observable, we only push a value when it exist and valid). But compiler in the case complains in the line this.accessTokenUpdated.next(this.accessToken) that it can not accept undefined value. Having a check like I have implemented removes this error.

Reply:
The problem was that `this.accessToken != undefined` is a typeguard, so it knows in the branch that token has to be string. By calling the function this typeguard is lost, that's why I casted it to string.